src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -1322,12 +1322,82 @@ public void LangVersion()
             Assert.Equal(defaultVersion, parsedArgs.ParseOptions.LanguageVersion);
         }
 
+        /// <summary>
+        /// When a new version is added, this test will break. This list must be checked:
+        /// - update the command-line error for bad /langver flag (<see cref="ErrorCode.ERR_BadCompatMode"/>)
+        /// - update the "UpgradeProject" codefixer
+        /// - update the IDE drop-down for selecting Language Version
+        /// - update the tests that include this canary
+        /// </summary>
+        [Fact]
+        public void LanguageVersionChanged_Canary()
+        {
+            AssertEx.SetEqual(new[] { "default", "1", "2", "3", "4", "5", "6", "7", "latest" },
+                Enum.GetValues(typeof(LanguageVersion)).Cast<LanguageVersion>().Select(v => v.ToDisplayString()));
+            // For minor versions, the format should be "x.y", such as "7.1"
+        }
+
         [Fact]
         public void RememberToUpdateDiagnosticsWhenUpdatingLangVersion()
         {
             // When new language versions are added, this test will fail. Remember to update the diagnostics message (ERR_BadCompatMode).
             Assert.Equal(LanguageVersion.CSharp7, LanguageVersion.Latest.MapSpecifiedToEffectiveVersion());
             Assert.Equal(LanguageVersion.CSharp7, LanguageVersion.Default.MapSpecifiedToEffectiveVersion());
+
+            LanguageVersionChanged_Canary();
+        }
+
+        [Fact]
+        public void LanguageVersion_MapSpecifiedToEffectiveVersion()
+        {
+            Assert.Equal(LanguageVersion.CSharp1, LanguageVersion.CSharp1.MapSpecifiedToEffectiveVersion());
```

Comment:
Thanks for the suggestion. I gave it a try, but I didn't feel it fit that well. 
One problem is that Default and Latest are irregularities, but it doesn't seem worth splitting the test or having two parameters. I'm tempted to leave as-is (seems simpler).
And there is also the canary that I want to include but not invoke many times.